src/test/java/org/broadinstitute/hellbender/tools/walkers/markduplicates/AbstractMarkDuplicatesCommandLineProgramTest.java

```diff
@@ -170,6 +167,30 @@ public void testOpticalDuplicateClusterSamePositionNoOpticalDuplicates() {
         tester.runTest();
     }
 
+    @Test
+    public void testOpticalDuplicatesDifferentReadGroups() {
+        final AbstractMarkDuplicatesTester tester = getTester();
+        tester.setExpectedOpticalDuplicate(0);
+        tester.addMatePair("RUNID:7:1203:2886:82292",  19, 19, 485253, 485253, false, false, true, true, "42M59S", "59S42M", true, false, false, false, false, DEFAULT_BASE_QUALITY, "H0164.2");  // duplicate
```

Comment:
I hate these 600 parameter methods... but I'm not sure what to do about them

Reply:
there are a bunch of overloads that help the problem. For the most par every field is necessary to at least one of these tests.